---
title: Condition-specific variants in Simphonia 500 RNA-seq
author: "Sylvain Schmitt sylvain.schmitt@agroparistech.fr"
date: '`r Sys.Date()`'
output:
  bookdown::html_document2:
    number_sections: false
    toc: true
    toc_float: yes
  bookdown::pdf_document2:
    number_sections: false
    toc: false
# csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
# bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(kissDE)
opts_chunk$set(echo = T, eval = T,
               message = F, warning = F, 
               fig.height = 4, fig.width = 8,
               cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/AREA/data/symphonia.trinity500.kissDE"
save_path <- "/media/toshiba/AREA/symphonia.trinity500.kissplice/kissDE.symphonia.trinity500/"
```

# Introduction

# Ecotypes

Define the ecotypes classes of each RNA-seq library. 37 individuals in total with pairEnd data containing 14 *Symphonia globulifera* and 23 *Symphonia sp1*.

```{r conditions}
conditions <- c(rep("glo",14),rep("sp1", 23))
```

# Analysis

```{r function}
analyse_DEV <- function(
  type, # only variant code e.g. 0a
  conditions, # conditions (ecotype)
  path = "~/Documents/BIOGECO/AREA/data/symphonia.trinity500.kissDE", # working directory
  pairedEnd = T, # kissplice2counts parameter
  pvalue = 0.001, # kissplice2counts parameter
  filterLowCountsVariants = 100 # kissplice2counts parameter
){
  file <- paste0("results_coherent_type_", type, ".fa")
  counts <- kissplice2counts(file.path(path, file), 
                                     pairedEnd = T)
  DEV <- diffExpressedVariants(counts$countsEvents, 
                               conditions, 
                               filterLowCountsVariants = filterLowCountsVariants,
                               pvalue = pvalue)
  return(DEV)
}
```


# Single SNPs 

```{r sSNPs}
# sSNPs <- analyse_DEV("0a", conditions)
# save(sSNPs, file = "./conditions-specific_save/sSNPs.Rdata")
load("./conditions-specific_save/sSNPs.Rdata")
writeOutputKissDE(sSNPs, file.path(save_path, "kissDE_output_type0a.tsv"))
```

# Multi SNPs

```{r mSNPs}
# mSNPs <- analyse_DEV("0b", conditions)
# save(mSNPs, file = "./conditions-specific_save/mSNPs.Rdata")
load("./conditions-specific_save/mSNPs.Rdata")
writeOutputKissDE(mSNPs, file.path(save_path, "kissDE_output_type0b.tsv"))
```

# Alternative splicing events

```{r altsplic}
# altsplic <- analyse_DEV("1", conditions)
# save(altsplic, file = "./conditions-specific_save/altsplic.Rdata")
load("./conditions-specific_save/altsplic.Rdata")
writeOutputKissDE(altsplic, file.path(save_path, "kissDE_output_type1.tsv"))
```

# Inexact tandem repeats

```{r msat}
# msat <- analyse_DEV("2", conditions)
# save(msat, file = "./conditions-specific_save/msat.Rdata")
load("./conditions-specific_save/msat.Rdata")
writeOutputKissDE(msat, file.path(save_path, "kissDE_output_type2.tsv"))
```

# short Indels

```{r sIndels}
# sIndels <- analyse_DEV("3", conditions)
# save(sIndels, file = "./conditions-specific_save/sIndels.Rdata")
load("./conditions-specific_save/sIndels.Rdata")
writeOutputKissDE(sIndels, file.path(save_path, "kissDE_output_type3.tsv"))
```

# All others

All others, with a shorter path > 2k not being a SNP

```{r others}
# others <- analyse_DEV("4", conditions)
# save(others, file = "./conditions-specific_save/others.Rdata")
load("./conditions-specific_save/others.Rdata")
writeOutputKissDE(others, file.path(save_path, "kissDE_output_type4.tsv"))
```

# Summary

```{r summary}
unlist(lapply(list("Single SNPs" = sSNPs, 
            "Multiple SNPs" = mSNPs, 
            "Alternative splicing events" = altsplic, 
            "Inexact tandem repeats" = msat,
            "short Indels" = sIndels,
            "Others" = others), 
       function(x) dim(x)[1])) %>%
  data.frame() %>% 
  kable()
```


# References
